tasks/lzards-backup/tests/test-index.js

```diff
@@ -0,0 +1,693 @@
+  const lzardsPostMethod = (async () => ({
+    body: 'failure body',
+    statusCode: 404,
+  }));
+  const roleCreds = { fake: 'creds_object' };
+  const name = 'fakeFilename';
+  const filepath = 'fakeFilePath';
+  const authToken = 'fakeToken';
+  const collectionId = 'FAKE_COLLECTION';
+  const bucket = 'fakeFileBucket';
+  const filename = 's3://fakeFileBucket/fakeFilename';
+
+  const file = {
+    name,
+    filepath,
+    bucket,
+    filename,
+  };
+  const granuleId = 'fakeGranuleId';
+
+  const actual = await index.makeBackupFileRequest({
+    authToken,
+    collectionId,
+    roleCreds,
+    file,
+    granuleId,
+    lzardsPostMethod,
+  });
+
+  const expected = {
+    body: 'failure body',
+    filename,
+    granuleId: 'fakeGranuleId',
+    status: 'FAILED',
+    statusCode: 404,
+  };
+
+  t.deepEqual(actual, expected);
+});
+
+test('makeBackupFileRequest returns expected makeBackupFileRequestResult on other failure', async (t) => {
+  const lzardsPostMethod = (async () => {
+    throw new Error('DANGER WILL ROBINSON');
+  });
+  const roleCreds = { fake: 'creds_object' };
+  const name = 'fakeFilename';
+  const filepath = 'fakeFilePath';
+  const authToken = 'fakeToken';
+  const collectionId = 'FAKE_COLLECTION';
+  const bucket = 'fakeFileBucket';
+  const filename = 's3://fakeFileBucket/fakeFilename';
+
+  const file = {
+    name,
+    filepath,
+    bucket,
+    filename,
+  };
+  const granuleId = 'fakeGranuleId';
+
+  let actual = await index.makeBackupFileRequest({
+    authToken,
+    collectionId,
+    roleCreds,
+    file,
+    granuleId,
+    lzardsPostMethod,
+  });
+
+  const expected = {
+    body: '{"name":"Error"}',
+    filename,
+    granuleId: 'fakeGranuleId',
+    status: 'FAILED',
+  };
+
+  actual = removeStackObjectFromErrorBody(actual);
+  t.deepEqual(actual, expected);
+});
+
+test('makeBackupFileRequest returns expected makeBackupFileRequestResult', async (t) => {
+  const accessUrl = 'fakeURL';
+  const generateAccessUrlMethod = (() => accessUrl);
+  const lzardsPostMethod = (async () => ({
+    body: 'fake body',
+    statusCode: 201,
+  }));
+
+  const roleCreds = { fake: 'creds_object' };
+  const name = 'fakeFilename';
+  const filepath = 'fakeFilePath';
+  const authToken = 'fakeToken';
+  const collectionId = 'FAKE_COLLECTION';
+  const bucket = 'fakeFileBucket';
+  const filename = 's3://fakeFileBucket/fakeFileBucket';
+
+  const file = {
+    name,
+    filepath,
+    bucket,
+    filename,
+  };
+  const granuleId = 'fakeGranuleId';
+
+  const actual = await index.makeBackupFileRequest({
+    authToken,
+    collectionId,
+    roleCreds,
+    file,
+    granuleId,
+    generateAccessUrlMethod,
+    lzardsPostMethod,
+  });
+
+  const expected = {
+    body: 'fake body',
+    filename,
+    granuleId: 'fakeGranuleId',
+    status: 'COMPLETED',
+    statusCode: 201,
+  };
+
+  t.deepEqual(actual, expected);
+});
+
+test('getGranuleCollection returns queried collections', async (t) => {
```

Comment:
I could be wrong, but I think we don't need this test. `getGranuleCollection()` just wraps `getCollection` with some logic to catch and throw errors if the necessary arguments are missing. So I think we only need to test that errors are thrown in the right cases, which you already have. Whether the right collection is returned is ultimately handled by `getCollection`, which already has its own test coverage.

Reply:
Perhaps the test should be titled "returns a queried collection"?  The test here is more that it makes the wrapped call, than the results (which are the default return mock).      While it *is* a wrapper, wrapping the call in a particular way is logic. 

I think the better argument for removal is that we're testing the method chain that is calling getGranuleCollection with a working call, and we shouldn't be using this method outside that chain,  which is fair.

Reply:
ef6a85150